### PR TITLE
Add gradle support for generating go files from proto and running tests

### PR DESCRIPTION
**Pull request by niloc132**
This patch will land in pieces upstream, so consider not merging it directly, I'm creating it this way to make it easier to review.

### CODE REVIEW

File: go-client/go-test-to-junit.sh
Comment (devinrsmith) [inline go-client/go-test-to-junit.sh:2]: I would start:

```
#!/usr/bin/env bash

set -o errexit
set -o pipefail
set -o nounset
set -o xtrace
...
```

File: go-client/build.gradle
Comment (devinrsmith) [inline go-client/build.gradle:33]: I know we use this elsewhere, but we should prefer to start moving to `tasks.named(...)` where applicable. New stuff prefer to follow that.

File: go-client/build.gradle
Comment (devinrsmith) [inline go-client/build.gradle:81]: `tasks.named('check')`

File: proto/proto-backplane-grpc/Dockerfile
Comment (devinrsmith) [inline proto/proto-backplane-grpc/Dockerfile:8]: What am I missing here - where are these files coming from? Should we extend protoc-base to include them?

File: py/client/build.gradle
Comment (devinrsmith) [inline py/client/build.gradle:60]: named

**Review by devinrsmith — COMMENTED** [review]
Quick gradle pass, things look ok. I haven't run these tasks yet; will want to before merging into core.

File: proto/proto-backplane-grpc/Dockerfile
Comment (niloc132) [reply proto/proto-backplane-grpc/Dockerfile:8]: Ah right, need to revisit that. I don't recall if we found a canonical place to get cross platform binaries from, I'll dig in.

The https://github.com/deephaven/deephaven-core/pull/2667 patch won't have this yet.

File: py/client/build.gradle
Comment (niloc132) [reply py/client/build.gradle:60]: I can't find a single instance where we use named with quick, since that requires a .configure closure

File: go-client/go-test-to-junit.sh
Comment (niloc132) [reply go-client/go-test-to-junit.sh:2]: Has this convention been updated elsewhere, documented so we can be sure to fix existing ones, future ones?

...and if we're changing the standard again, is it worth changing it further?

File: go-client/build.gradle
Comment (niloc132) [reply go-client/build.gradle:32]: ignoring these, as per slack discussion.

File: go/build.gradle
Comment (devinrsmith) [inline go/build.gradle:73]: It might be worth it to separate out the dependencies we need to download / install before copying over the full project. Essentially, would be nice to be able to edit locally w/o needing to re-install all of the dependencies.

File: docker/registry/go/gradle.properties
Comment (devinrsmith) [inline docker/registry/go/gradle.properties:2]: I think we should prefer to track `golang:1`. (`1.18.4` is latest available.) If a golang update `1.19` causes the build to break, I can explicitly track `1.18` at that point in time.

File: proto/proto-backplane-grpc/build.gradle
Comment (devinrsmith) [inline proto/proto-backplane-grpc/build.gradle:64]: Not needed anymore, right?

File: go/build.gradle
Comment (devinrsmith) [inline go/build.gradle:34]: Nothing references `update`. Mistake, or don't need variable?

**Review by devinrsmith — COMMENTED** [review]
I've got some dependency issues still after cleaning protobuf. Will discuss in Slack.

File: go/build.gradle
Comment (devinrsmith) [reply go/build.gradle:34]: I see py-client has same pattern.

File: go/build.gradle
Comment (devinrsmith) [inline go/build.gradle:50]: I've tried to get this task marked as `shouldRunAfter compare` - I think we should arguably do the compare as early on as possible, but it didn't seem to work. We may really mean the *first* or *all* tasks of `registerDockerTask`. Instead of the returned one. Not a blocker to merge, but may be a nicety we can do if easy.

Maybe there is really something more general we should add to our setup: `quick` tasks should preferably be run before other check/test tasks.

File: proto/proto-backplane-grpc/Dockerfile
Comment (niloc132) [inline proto/proto-backplane-grpc/Dockerfile:4]: could also consider moving these to our images. 

worth doing now, or wait until we land the go client api patch, and then make me fix it?

File: proto/proto-backplane-grpc/build.gradle
Comment (niloc132) [reply proto/proto-backplane-grpc/build.gradle:64]: Correct, will remove.

File: go/build.gradle
Comment (niloc132) [reply go/build.gradle:73]: Given the mod download, I don't feel too bad about listing go-junit-report as something to download.

Comment (devinrsmith) [reply go/build.gradle:73]: I mean, I think we should try and get mod download `RUN` before we copy the remaining `/project`.

Comment (niloc132) [reply go/build.gradle:73]: I can move the line of code, but can't tell you what the implications of doing so are, if it could potentially have an impact without our go.mod/go.sum copied and potentially managing dependency versions.

File: docker/registry/go/gradle.properties
Comment (niloc132) [reply docker/registry/go/gradle.properties]: Decided in slack that we're going to assume 1.x here.

**Review by devinrsmith — APPROVED** [review]

**Pull request closed without merging**
